configuration/exodus-lambda-deploy.yaml

```diff
@@ -0,0 +1,148 @@
+    MinLength: 10
+    Description: The origin access identity ID associated with the environment
+
+Resources:
+  Distribution:
+    Type: AWS::CloudFront::Distribution
+    Properties:
+      DistributionConfig:
+        Comment: !Sub exodus-cdn-${env}
+        DefaultCacheBehavior:
+          AllowedMethods:
+            - GET
+            - HEAD
+          CachedMethods:
+            - GET
+            - HEAD
+          ForwardedValues:
+            QueryString: false
+            Cookies:
+              Forward: none
+          LambdaFunctionAssociations:
+            - EventType: origin-request
+              LambdaFunctionARN: !Ref OriginRequestFunc.Version
+            - EventType: origin-response
+              LambdaFunctionARN: !Ref OriginResponseFunc.Version
+          TargetOriginId: !Sub S3-exodus-cdn-${env}
+          ViewerProtocolPolicy: redirect-to-https
+        Enabled: true
+        HttpVersion: http2
+        Origins:
+          - DomainName: !Sub "exodus-cdn-${env}.s3.amazonaws.com"
+            Id: !Sub "S3-exodus-cdn-${env}"
+            S3OriginConfig:
+              OriginAccessIdentity: !Sub "origin-access-identity/cloudfront/${oai}"
+
+  OriginRequestFunc:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub origin-request-${env}
+      CodeUri:
+      Handler: exodus_lambda.origin_request
+      Role: !GetAtt FuncRole.Arn
+      Runtime: python3.7
+      Timeout: 5
+      AutoPublishAlias: live
+
+  OriginResponseFunc:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub origin-response-${env}
+      CodeUri:
+      Handler: exodus_lambda.origin_response
+      Role: !GetAtt FuncRole.Arn
+      Runtime: python3.7
+      Timeout: 5
+      AutoPublishAlias: live
+
+  FuncRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Sub exodus-lambda-${env}
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - sts:AssumeRole
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+                - edgelambda.amazonaws.com
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AWSLambdaFullAccess
+        - arn:aws:iam::aws:policy/CloudFrontFullAccess
+        - arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess
+        - arn:aws:iam::aws:policy/AWSCodePipelineFullAccess
+        - arn:aws:iam::aws:policy/AWSLambdaInvocation-DynamoDB
+      Path: /
+
+  OriginRequestAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: !Sub origin-request-alarm-${env}
+      AlarmDescription: !Sub origin-request-${env} invocation errors
+      Namespace: AWS/Lambda
+      MetricName: Errors
+      Dimensions:
+        - Name: FunctionName
+          Value: !Ref OriginRequestFunc
+      Statistic: Sum
+      Period: 180
+      EvaluationPeriods: 1
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      Threshold: 3
```

Comment:
I have little idea if this is correct.  I won't be surprised if, at the scale we're expecting, we get many many more errors than this all the time - given that clients can probably trigger errors by bad behavior on their end.  Anyway I don't see much choice but to start with a guess and refine it over time.

Reply:
You're probably right. The idea behind this threshold was for it to cause a rollback of the CloudFormation stack during validation. So, a few failed pipeline tests should be enough to keep out bad code.